src/wallet/wallet.h

```diff
@@ -177,28 +178,33 @@ class CMerkleTx : public CTransaction
      */
     int nIndex;
 
-    CMerkleTx()
+    template<typename... Args>
+    CMerkleTx(Args&&... args)
     {
+        SetTx(std::forward<Args>(args)...);
         Init();
     }
 
-    CMerkleTx(const CTransaction& txIn) : CTransaction(txIn)
-    {
-        Init();
-    }
+    operator const CTransaction&() const { return *tx; }
 
     void Init()
     {
         hashBlock = uint256();
         nIndex = -1;
     }
 
+    template<typename... Args>
+    void SetTx(Args&&... args)
```

Comment:
This `std::forward` stuff is interesting because it means that the `SetTx` method and the `CMerkleTx` and `CWalletTx` constructors can all be called with arguments for any of the 3 non-template overloads of `MakeTransactionRef`, or with arguments for any of the 4 overloaded `CTransaction` constructors. But I wonder if you would consider defining `SetTx` more simply as:

```
void SetTx(CTransactionRef ref)
{
    tx = std::move(ref);
}
```

and similarly replacing the template constructors with regular constructors accepting `CTransactionRef`s.

The advantage of doing this for someone reading the code is that you won't have to look 3 levels deep to see what arguments a `SetTx` call or constructor will accept, and of course won't have to know anything about std::forward either.